great_expectations/agent/agent.py

```diff
@@ -0,0 +1,159 @@
+            subscriber.consume(
+                queue=self._config.organization_id,
+                on_message=self._handle_event_as_thread,
+            )
+        except KeyboardInterrupt:
+            print("Received request to shutdown.")
+        except (SubscriberError, ClientError) as e:
+            print("Connection to GX Cloud has encountered an error.")
+            print("Please restart the agent and try your action again.")
+            print(e)
+        finally:
+            self._close_subscriber(subscriber)
+
+    def _handle_event_as_thread(self, event: Event, correlation_id: str) -> None:
+        """Schedule _handle_event to run in a thread.
+
+        Callback passed to Subscriber.consume which forwards events to
+        the EventHandler for processing.
+
+        Args:
+            event: pydantic model representing an event
+            correlation_id: stable identifier for an event across its lifecycle
+        Raises:
+            RequeueMessageError when the worker is busy
+        """
+
+        if self._can_accept_new_task() is not True:
+            # signal to Subscriber that we can't process this message right now
+            raise RequeueMessageError
+        self._current_task = self._executor.submit(
+            self._handle_event, event=event, correlation_id=correlation_id
+        )
+
+    def _handle_event(self, event: Event, correlation_id: str) -> None:
+        """Pass events to EventHandler.
+
+        Callback passed to Subscriber.consume which forwards events to
+        the EventHandler for processing.
+
+        Args:
+            event: pydantic model representing an event
+            correlation_id: stable identifier for an event across its lifecycle
+        """
+        # TODO lakitu-139: record job as started
+
+        handler = EventHandler(context=self._context)
+        handler.handle_event(event=event, correlation_id=correlation_id)
+
+        # TODO lakitu-139: record job as complete
+        return
+
+    def _can_accept_new_task(self) -> bool:
+        return self._current_task is None or self._current_task.done()
+
+    def _close_subscriber(self, subscriber: Optional[Subscriber]) -> None:
+        """Ensure the subscriber has been closed."""
+        if subscriber is None:
+            return  # nothing to close
+        try:
+            subscriber.close()
+        except SubscriberError as e:
+            print("Subscriber encountered an error while closing:")
+            print(e)
+
+    @classmethod
+    def _get_config_from_env(cls) -> GXAgentConfig:
+        """Construct GXAgentConfig from available environment variables"""
+        url = os.environ.get("BROKER_URL", None)
+        org_id = os.environ.get("GE_CLOUD_ORGANIZATION_ID", None)
+        try:
+            return GXAgentConfig(
+                organization_id=org_id, broker_url=url
+            )  # noqa - mypy doesnt know pydantic handles url type
```

Comment:
This is the wrong format for ignoring mypy warning.
https://mypy.readthedocs.io/en/stable/error_codes.html#silencing-errors-based-on-error-codes